### ISSUE TITLE
Allow disabling fetching of test outputs of tests running remotely

### ISSUE THREAD

**Issue by sfc-gh-sgiesecke** [bazelbuild/bazel#20068]
### Description of the feature request:

We would like to be able to disable fetching of test outputs of tests running remotely, in particular of tests that PASSED.

AFAIU, this is similar in spirit but not the same as #19782

### Which category does this issue belong to?

Performance, Remote Execution

### What underlying problem are you trying to solve with this feature?

Test outputs can be huge and take a long time to download.

### Which operating system are you running Bazel on?

Linux

**Comment by coeuvre** [comment]
Can you give an example of the test outputs you don't want Bazel to download? Is it outputs inside undeclared test output directory?

**Comment by sfc-gh-sgiesecke** [comment]
> Is it outputs inside undeclared test output directory?

I think so, yes.

**Comment by tjgq** [comment]
Do I understand it correctly that you'd like Bazel to decide whether to download test outputs depending on whether the test passed or failed (download for failing, but not for passing)?

**Comment by sfc-gh-sgiesecke** [comment]
I'd like to have an option that controls this. Like `--download_test_outputs={none,failing,all}`. The current behaviour would be equivalent to `all`. With `failing`, the behaviour would be as you described. With `none`, test outputs would never be downloaded, even for failing tests. I guess implementing the latter as well would be rather trivial, so it would make sense to have that. If it's not trivial then `all` and `failing` would be sufficient, too.

**Comment by coeuvre** [comment]
I don't think we should add more flags to control which output to download or not -- that's a rabbit hole and probably more flags will be needed later for different cases.

The tradeoff we want to make is: BwoB is easy to use (which implies no need to tune many flags), giving performance wins for most cases but not perfect -- there will be cases when skip downloading intermediate outputs are not desired or toplevel outputs are unnecessarily downloaded.

To have a perfect lazily download behaviour, use output service (still WIP, https://github.com/bazelbuild/bazel/pull/12823) instead, which requires some effort to set up.

**Comment by sfc-gh-sgiesecke** [comment]
> To have a perfect lazily download behaviour, use output service (still WIP, https://github.com/bazelbuild/bazel/pull/12823) instead, which requires some effort to set up.

We're turning circles here, we've been there and back.

**Comment by coeuvre** [comment]
To be clear, we are actively working on #12823 and it will be merged soon (with some modifications).
